### ISSUE TITLE
Python unit test failure in test_json

### ISSUE THREAD

**Issue by znation** [apple/turicreate#992]
Reproduced on Linux in Python 2.7; may reproduce elsewhere.

```
_________________________ JSONTest.test_malformed_json _________________________

self = <turicreate.test.test_json.JSONTest testMethod=test_malformed_json>

    def test_malformed_json(self):
>       self.assertRaises(IOError, SArray.read_json, './malformed.json')

test_json.py:254: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../data_structures/sarray.py:585: in read_json
    proxy.load_from_json_record_files(_make_internal_url(filename))
turicreate/cython/cy_sarray.pyx:72: in turicreate.cython.cy_sarray.UnitySArrayProxy.load_from_json_record_files
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   RuntimeError: Unable to parse /builds/turi/turicreate/deps/env/lib/python2.7/site-packages/turicreate/test/malformed.json. It does not appear to be in JSON record format. A list of dictionaries is expected

turicreate/cython/cy_sarray.pyx:75: RuntimeError
----------------------------- Captured stdout call -----------------------------
Parsing JSON records from /builds/turi/turicreate/deps/env/lib/python2.7/site-packages/turicreate/test/malformed.json
--------------------------- Captured stdout teardown ---------------------------
```

**Comment by TobyRoseman** [comment]
Looks related to #967

**Comment by znation** [comment]
@TobyRoseman Agreed, it does. Assigning to @hoytak.

**Comment by znation** [comment]
@hoytak I'm not sure what you mean re: "old test" and "new version". This failure is from a single commit of the code (`f6840ebb45394140db65db74cc031eafdaeec075`), built and running tests.

**Comment by znation** [comment]
Fixed with #1010.
